test/clj_yaml/core_test.clj

```diff
@@ -199,3 +200,23 @@ the-bin: !!binary 0101")
   (testing "emoji in comments are OK too"
     (let [yaml "# 💣 emoji in a comment\n42"]
       (is (= 42 (parse-string yaml))))))
+
+(def too-many-aliases
+  (->> (range 51)
+       (map #(str "b" % ": *a"))
+       (cons "a: &a [\"a\",\"a\"]")
+       (string/join "\n")))
+
+(deftest max-aliases-for-collections-works
+  (is (thrown? YAMLException (parse-string too-many-aliases)))
+  (is (parse-string too-many-aliases :max-aliases-for-collections 51)))
+
+(def recursive-yaml "
+---
+&A
+- *A: *A
+")
+
+(deftest allow-recursive-works
+  (is (thrown? YAMLException (parse-string recursive-yaml)))
```

Comment:
Let's change this assertion to `thrown-with-msg?` and check that the exception has the word `Recursive` in the message.

https://bitbucket.org/asomov/snakeyaml/src/1922347d615264c50041223ec2bd689906cbcb26/src/main/java/org/yaml/snakeyaml/constructor/BaseConstructor.java?at=master#lines-484